core/modules/memman/MemFile.h

```diff
@@ -0,0 +1,143 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2016 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_QSERV_MEMMAN_MEMFILE_H
+#define LSST_QSERV_MEMMAN_MEMFILE_H
+
+// Qserv headers
+#include "memman/Memory.h"
+
+// System headers
+#include <atomic>
+#include <mutex>
+#include <string>
+#include <unistd.h>
+
+namespace lsst {
+namespace qserv {
+namespace memman {
+
+//-----------------------------------------------------------------------------
+//! @brief Description of a memory based file.
+//-----------------------------------------------------------------------------
+
+class MemFile {
+public:
+
+    //-----------------------------------------------------------------------------
+    //! @brief Get file path for this table.
+    //!
+    //! @return The file path.
+    //-----------------------------------------------------------------------------
+
+    std::string filePath() {return _fPath;}
```

Comment:
I hate returning references to things inside an object that may be later deleted. In this case, I will just remove the method as it's not used anywhere in the code.